data/interfaces/default/newsletter_config.html

```diff
@@ -471,8 +486,201 @@ <h4 class="modal-title" id="newsletter-config-modal-header">${newsletter['agent_
+                        $btn.find('i').attr('class', originalIcon);
+                        $btn.attr('title', originalTitle);
+                    }, 2000);
+                } finally {
+                    $btn.prop('disabled', false);
+                }
+            }
+        });
+        
+        // Toggle word wrap button
+        $('#toggle-wrap').off('click').on('click', function() {
+            if (messageEditor) {
+                var currentWrap = messageEditor.getSession().getUseWrapMode();
+                messageEditor.getSession().setUseWrapMode(!currentWrap);
+            }
+        });
+        
+        // Toggle gutter button
+        $('#toggle-gutter').off('click').on('click', function() {
+            if (messageEditor) {
+                var currentGutter = messageEditor.renderer.getShowGutter();
+                messageEditor.renderer.setShowGutter(!currentGutter);
+            }
+        });
+    }
+    
+
+    
+    function formatHTML(html) {
+        // Clean up the input first
+        var cleanHtml = html.trim();
+        if (!cleanHtml) {
+            return cleanHtml;
+        }
+        
+        try {
+            // Check if content looks like HTML
+            if (cleanHtml.includes('<') && cleanHtml.includes('>')) {
+                return html_beautify(cleanHtml, {
+                    indent_size: 4,
+                    indent_char: ' ',
+                    max_preserve_newlines: 5,
+                    preserve_newlines: true,
+                    keep_array_indentation: false,
+                    break_chained_methods: false,
+                    indent_scripts: 'normal',
+                    brace_style: 'collapse',
+                    space_before_conditional: true,
+                    unescape_strings: false,
+                    jslint_happy: false,
+                    end_with_newline: false,
+                    wrap_line_length: 0,
+                    indent_inner_html: false,
+                    comma_first: false,
+                    e4x: false,
+                    indent_empty_lines: false
+                });
+            } else {
+                // If it's not HTML, try to format as plain text
+                return html_beautify('<div>' + cleanHtml + '</div>', {
+                    indent_size: 2,
+                    indent_char: ' ',
+                    max_preserve_newlines: 2,
+                    preserve_newlines: true
+                }).replace(/^<div>/, '').replace(/<\/div>$/, '');
+            }
+        } catch (error) {
+            console.warn('js-beautify failed', error);
+        }
+    }
+
+    
+    $(document).ready(function() {
+        // Initialize Ace editor when the modal is shown
+        $('#newsletter-config-modal').on('shown.bs.modal', function() {
+            setTimeout(function() {
+                if ($('#message-editor').length && !messageEditor) {
+                    initMessageEditor();
+                }
+            }, 100);
+        });
```

Comment:
Initialize the editor when the tab is shown.
```suggestion
        $('#newsletter-text-tab').on('shown.bs.tab', function() {
            initMessageEditor();
        });
```

Add `id="newsletter-text-tab"` to the tab link on line 25